cpp/src/community/k_truss_impl.cuh

```diff
@@ -114,246 +188,547 @@ k_truss(raft::handle_t const& handle,
 
+  cugraph::edge_property_t<graph_view_t<vertex_t, edge_t, false, multi_gpu>, bool> dodg_mask(
+    handle, cur_graph_view);
+  {
     auto out_degrees = cur_graph_view.compute_out_degrees(handle);
-    edge_src_property_t<decltype(cur_graph_view), edge_t> edge_src_out_degrees(handle,
-                                                                               cur_graph_view);
-    edge_dst_property_t<decltype(cur_graph_view), edge_t> edge_dst_out_degrees(handle,
-                                                                               cur_graph_view);
     update_edge_src_property(
       handle, cur_graph_view, out_degrees.begin(), edge_src_out_degrees.mutable_view());
     update_edge_dst_property(
       handle, cur_graph_view, out_degrees.begin(), edge_dst_out_degrees.mutable_view());
 
-    rmm::device_uvector<vertex_t> srcs(0, handle.get_stream());
-    rmm::device_uvector<vertex_t> dsts(0, handle.get_stream());
+    cugraph::fill_edge_property(
+      handle, unmasked_cur_graph_view, dodg_mask.mutable_view(), bool{false});
+
+    cugraph::transform_e(
+      handle,
+      cur_graph_view,
+      edge_src_out_degrees.view(),
+      edge_dst_out_degrees.view(),
+      edge_dummy_property_t{}.view(),
+      [] __device__(auto src, auto dst, auto src_out_degree, auto dst_out_degree, auto) {
+        return (src_out_degree < dst_out_degree) ? true
+               : ((src_out_degree == dst_out_degree) &&
+                  (src < dst) /* tie-breaking using vertex ID */)
+                 ? true
+                 : false;
+      },
+      dodg_mask.mutable_view(),
+      false);
+
+    if (cur_graph_view.has_edge_mask()) { cur_graph_view.clear_edge_mask(); }
+    cur_graph_view.attach_edge_mask(dodg_mask.view());
+  }
 
-    edge_weight_view =
-      edge_weight ? std::make_optional((*edge_weight).view())
-                  : std::optional<edge_property_view_t<edge_t, weight_t const*>>{std::nullopt};
-    if (edge_weight_view) {
-      std::tie(srcs, dsts, wgts) = extract_transform_e(
-        handle,
-        cur_graph_view,
-        edge_src_out_degrees.view(),
-        edge_dst_out_degrees.view(),
-        *edge_weight_view,
-        extract_low_to_high_degree_weighted_edges_t<vertex_t, weight_t, edge_t>{});
-    } else {
-      std::tie(srcs, dsts) =
+  // 4. Compute triangle count using nbr_intersection and unroll weak edges
+
+  {
+    cugraph::edge_property_t<decltype(cur_graph_view), bool> weak_edges_mask(
+      handle, unmasked_cur_graph_view);
+    cugraph::fill_edge_property(
+      handle, unmasked_cur_graph_view, weak_edges_mask.mutable_view(), bool{true});
+
+    // Mask self loops and edges not being part of k-1 core
+    cugraph::transform_e(
+      handle,
+      unmasked_cur_graph_view,
+      edge_src_dummy_property_t{}.view(),
+      edge_dst_dummy_property_t{}.view(),
+      edge_mask.view(),
+      [] __device__(auto src, auto dst, auto src_out_degree, auto dst_out_degree, auto e_mask) {
```

Comment:
```
auto src_out_degree, auto dst_out_degree,
```
=>
```
auto, auto,
```
`src_out_degree` & `dst_out_degree` are  misnomers... They are dummy values.

Reply:
Right, I am not passing anything for the src and dat property.